enhancements/network/ovs-dynamic-cpu-pinning.md

```diff
@@ -0,0 +1,201 @@
+  on every node where it wants to enable the pinning feature.
+- The file is used by ovnkube-node pod to start a goroutine that constantly monitors 
+  the CPUs where ovnkube-node itself runs and where OVS daemon run, aligning the latter if 
+  needed (see [CPU affinity alignment](#cpu-affinity-alignment)). The file is the signal for ovnk to enable 
+  the feature (see [Activation file](#activation-file))
+
+```mermaid
+sequenceDiagram
+    Note right of NodeTuningOperator: via MachineConfig
+  
+    NodeTuningOperator->>ovs-vswitchd: set_slice
+    NodeTuningOperator->>ovsdb-server: set_slice
+    NodeTuningOperator->>/etc/openvswitch/enable_dynamic_cpu_affinity: fwrite('1')
+    activate /etc/openvswitch/enable_dynamic_cpu_affinity
+    ovnkube-node ->> /etc/openvswitch/enable_dynamic_cpu_affinity: fstat()
+    
+    activate ovnkube-node
+    loop
+    ovnkube-node ->> ovnkube-node: x = sched_getaffinity(x)
+    ovnkube-node ->> ovs-vswitchd: sched_setaffinity(x)
+    ovnkube-node ->> ovsdb-server: sched_setaffinity(x)
+    end
+
+    deactivate ovnkube-node
+    deactivate /etc/openvswitch/enable_dynamic_cpu_affinity
+```
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+#### CPU affinity alignment
+
+CPU affinity alignment is implemented using system calls SYS_SCHED_SETAFFINITY and SYS_SCHED_GETAFFINITY.
+Pods from `ovnkube-node` daemonset run in the Burstable Quality of Service (QoS) class, hence their CPU 
+affinities are managed by the [kubelet cpu manager](https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/).
+
+#### Activation file
+
+The file `/etc/openvswitch/enable_dynamic_cpu_affinity` on the host filesystem is used to 
+activate the feature on ovnkube-node according to the following logic:
+a. If the file is missing OR the file file is empty, then the feature is turned `OFF`.
+b. If the file exists and its content is not an empty string, then the feature is turned `ON`.
+c. `ovnkube-node` responds to the file modifications (deletion or content update)
+  at runtime, without the need for a restart.
+
+#### Must-Gather enhancement
+
+As the feature is enabled via a file on the host filesystem, it's useful to
+add `/etc/openvswitch/enable_dynamic_cpu_affinity` file contents to the [gather_network_logs](https://github.com/openshift/must-gather/blob/master/collection-scripts/gather_network_logs) script.
+
+### Risks and Mitigations
+
+|Risk|Mitigation|
+|---|---|
+|After NTO enabled the feature, a network performance decrease occurred in some cluster nodes.|The cluster administrator can turn off the feature on specific nodes by connecting through SSH and deleting the activation file. An overriding MachineConfig can be created to achieve the same goal.|
+
+
+### Drawbacks
+
+This enhancement breaks the hard division between `reservedSystemCPUs` list and CPUs designated
+to the Kubernetes workload.
```

Comment:
Only for the OVS service.

Reply:
In fact, this creates a hybrid between OCP 3.x where OVS was fully running as a workload (which caused reboot and upgrade issues) and OCP 4.x where OVS is too restricted.